tests/integration/file_append_transaction_e2e_test.py

```diff
@@ -0,0 +1,320 @@
+    required_chunks = file_append_tx.get_required_chunks()
+    assert required_chunks > 1, f"Expected multiple chunks, got {required_chunks}"
+    
+    append_receipt = file_append_tx.execute(env.client)
+    assert append_receipt.status == ResponseCode.SUCCESS, f"Append multiple chunks failed with status: {ResponseCode(append_receipt.status).name}"
+
+@mark.integration
+def test_integration_file_append_transaction_custom_chunk_settings(env):
+    """Test file append with custom chunk size and interval settings."""
+    operator_key = env.operator_key.public_key()
+
+    # Create a file first
+    create_receipt = (
+        FileCreateTransaction()
+        .set_keys(operator_key)
+        .set_contents(b"")
+        .execute(env.client)
+    )
+    
+    assert create_receipt.status == ResponseCode.SUCCESS, f"Create file failed with status: {ResponseCode(create_receipt.status).name}"
+    file_id = create_receipt.fileId
+
+    # Test with custom chunk settings
+    test_content = generate_uint8_array(5000)  # 5KB content
+    
+    append_tx = (
+        FileAppendTransaction()
+        .set_file_id(file_id) 
+        .set_contents(test_content)
+        .set_chunk_size(1000)  # 1KB chunks
+        .set_max_chunks(10)  # Max 10 chunks
+    )
+    
+    # Verify settings are applied
+    assert append_tx.chunk_size == 1000
+    assert append_tx.max_chunks == 10
+    
+    # Should require 5 chunks (5000 bytes / 1000 bytes per chunk)
+    required_chunks = append_tx.get_required_chunks()
+    assert required_chunks == 5, f"Expected 5 chunks, got {required_chunks}"
+    
+    append_receipt = append_tx.execute(env.client)
+    assert append_receipt.status == ResponseCode.SUCCESS, f"Append with custom settings failed with status: {ResponseCode(append_receipt.status).name}"
+
+@mark.integration
+def test_integration_file_append_transaction_max_chunks_exceeded(env):
+    """Test that FileAppendTransaction fails when max chunks is exceeded."""
+    operator_key = env.operator_key.public_key()
+
+    # Create a file first
+    create_receipt = (
+        FileCreateTransaction()
+        .set_keys(operator_key)
+        .set_contents(b"")
+        .execute(env.client)
+    )
+    
+    assert create_receipt.status == ResponseCode.SUCCESS
+    file_id = create_receipt.fileId
+
+    # Create content that would require more than max allowed chunks
+    large_content = generate_uint8_array(10000)  # 10KB content
+    
+    append_tx = (
+        FileAppendTransaction()
+        .set_file_id(file_id)
+        .set_contents(large_content)
+        .set_chunk_size(100)  # Small chunks = 100 chunks needed
+        .set_max_chunks(5)    # But only allow 5 chunks
+    )
+    
+    # Should fail with max chunks exceeded
+    try:
+        append_receipt = append_tx.execute(env.client)
+        assert False, "Expected ValueError for too many chunks"
+    except ValueError as e:
+        assert "more than" in str(e) and "chunks" in str(e)
```

Comment:
I would suggest doing something like this:

```suggestion
    with pytest.raises(ValueError, match="more than.*chunks"):
         append_receipt = append_tx.execute(env.client)
```